### PR TITLE
✨1440 - init filters context

### PR DESCRIPTION
**Pull request by anncatton**
**Description of changes**

Adds FiltersContext to File Repository
Integrates filters with url state
Uses mock filter button and sqon values for testing, will be connected to facets/search in next PR
Small change to NavBar login redirect to fix bug that prevented login from repository with filters in url

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: global/hooks/useFiltersContext.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  setFilters: (filter: FiltersType) => null,
+});
+
+export function FiltersProvider({ children }) {
+  const router = useRouter();
+  const currentQuery = getParams(router);
+
+  const [filters, setCurrentFilters] = React.useState(
+    JSON.parse(currentQuery.filters || null) || defaultFilters,
+  );
```
Comment (hlminh2000) [inline global/hooks/useFiltersContext.tsx:75]: why is this state necessary? `useUrlParamState` already provides a react state binding so we can just operate on that directly can't we?

File: global/hooks/useFiltersContext.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+export type FiltersType = {
+  op: CombinationKeys;
+  content: Operator[];
+};
+
+type FiltersContextType = {
+  filters: FiltersType;
+  clearFilters: () => void;
+  setFilters: (FiltersType) => void;
+};
```
Comment (hlminh2000) [inline global/hooks/useFiltersContext.tsx:59]: is this stuff accessible from `sqon-builder`? looks like we're really just interested in `FiltersType` eh

Comment (anncatton) [reply global/hooks/useFiltersContext.tsx:59]: Yeah we would just need to complete type...but the types aren't exported atm. Would that be worth a change in `sqon-builder`?

Comment (hlminh2000) [reply global/hooks/useFiltersContext.tsx:59]: I see :P yeah it'd be a nice thing, but not necessary right now :) just wanted to check

**Comment by anncatton** [comment]
> I don't think the file `global/hooks/useFiltersContext.tsx` is really meant for application-wide reuse eh, might want to use have it defined somewhere in `components/pages/file-repository` cause it's really only relevant for the file repo

@rosibaj said the filters may be carried through to some other pages in future, like we have on some of the gdc entity pages. So maybe keep in global?

**Comment by rosibaj** [comment]
> > I don't think the file `global/hooks/useFiltersContext.tsx` is really meant for application-wide reuse eh, might want to use have it defined somewhere in `components/pages/file-repository` cause it's really only relevant for the file repo
> 
> @rosibaj said the filters may be carried through to some other pages in future, like we have on some of the gdc entity pages. So maybe keep in global?

Happy to have call to discuss future business cases if needed!

File: global/hooks/useFiltersContext.tsx (hunk 1 above)
Comment (anncatton) [reply global/hooks/useFiltersContext.tsx:75]: yes i misunderstood all the things the param state was providing! reinvented the wheel a bit :)

**Pull request merged**